packages/snaps-execution-environments/src/webview/WebViewExecutorStream.ts

```diff
@@ -65,12 +64,11 @@ export class WebViewExecutorStream extends BasePostMessageStream {
   //  ignore.
   // eslint-disable-next-line no-restricted-syntax
   private _onMessage(event: PostMessageEvent): void {
-    if (!Array.isArray(event.data)) {
+    if (typeof event.data !== 'string') {
       return;
     }
 
-    const json = bytesToString(new Uint8Array(event.data as number[]));
-    const message = JSON.parse(json);
+    const message = JSON.parse(event.data);
```

Comment:
this can throw, should we catch gracefully?

Reply:
I don't think it matters too much, we don't do it in any of the other post message stream implementations used in MM either 😄